sdk/src/prover.rs

```diff
@@ -59,9 +50,15 @@ impl DevModeProver {
     }
 }
 
+/// Serializes the given input as a `Vec<u8>` compatible with Bonsai.
+fn serialize_input_to_bytes(input: impl serde::Serialize) -> Result<Vec<u8>> {
+    let input_encoded = risc0_zkvm::serde::to_vec(&input)?;
+    Ok(bytemuck::allocation::cast_vec(input_encoded))
```

Comment:
I believe the test is now failing because of this change

```suggestion
    Ok(bytemuck::cast_slice(&input_encoded).to_vec())
```

Reply:
Oh! Indeed